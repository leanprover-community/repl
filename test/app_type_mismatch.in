{"cmd": "example : 1 = 0 := by\n  cases 1\n  rfl\n  apply ?succ"}

{"cmd": "example : 1 = 0 := sorry", "sorries": "individual"}

{"tactic": "cases 1", "proofState": 0}

{"tactic": "rfl", "proofState": 1}

{"tactic": "apply ?succ", "proofState": 2}

