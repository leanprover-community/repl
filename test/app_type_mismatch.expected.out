{"messages":
 [{"severity": "error",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "(kernel) declaration has metavariables '_example'"}],
 "env": 0}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 19},
   "parentDecl": "_example",
   "goals": ["⊢ 1 = 0"],
   "endPos": {"line": 1, "column": 24}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 0},
   "endPos": {"line": 1, "column": 7},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 1,
 "goals": ["case zero\n⊢ 0 = 0", "case succ\nn✝ : Nat\n⊢ n✝ + 1 = 0"]}

{"proofStatus": "Incomplete: open goals remain",
 "proofState": 2,
 "goals": ["case succ\nn✝ : Nat\n⊢ n✝ + 1 = 0"]}

{"proofStatus": "Incomplete: contains metavariable(s)",
 "proofState": 3,
 "goals": []}

